### PR TITLE
Use appropriate go version according to the tag

### PR DESCRIPTION
**Pull request by hellozee**

### CODE REVIEW

File: tetrateci/setup_boring_go.sh [hunk 1]
```diff
@@ -1,13 +1,28 @@
-#!/bin/bash
+#!env bash
+set -e
+
+if $(grep "1.7" <<< $TAG ); then
```
Comment (deva26) [inline tetrateci/setup_boring_go.sh:4]: Can you check if these 3 go versions are available in boring_go?

Comment (hellozee) [reply tetrateci/setup_boring_go.sh:4]: Yes they do, :)

File: tetrateci/setup_go.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!env bash
+set -e
+
+echo "Fetching FIPS compliant Go"
```
Comment (deva26) [inline tetrateci/setup_go.sh:4]: Why FIPS compliant GO?

Comment (hellozee) [reply tetrateci/setup_go.sh:4]: sorry, I should have changed that

File: .github/workflows/make_release.yml [hunk 3]
```diff
@@ -32,13 +32,6 @@ jobs:
...
           password: ${{ secrets.BINTRAY_API_KEY }}
 
-      - name: set up go 1.15
-        uses: actions/setup-go@v1
-        # cause we setup go ourselves otherwise
-        if: ${{ !contains(github.event.ref, 'fips') }}
-        with:
-          go-version: 1.15
-
       - name: Run release builder
```
Comment (deva26) [inline .github/workflows/make_release.yml:35]: Is setting up regular GO added to create_release script?

File: .github/workflows/e2e_tests.yml [hunk 4]
```diff
@@ -303,13 +288,6 @@ jobs:
...
         shell: bash
 
-      - name: set up go 1.15
-        uses: actions/setup-go@v1
-        # cause we setup go ourselves otherwise
-        if: ${{ !contains(github.event.ref, 'fips') }}
-        with:
-          go-version: 1.15
-
       - name: run release builder
```
Comment (deva26) [inline .github/workflows/e2e_tests.yml:291]: Is setting up regular GO added to create_release script?

File: .github/workflows/make_release.yml (hunk 3 above)
Comment (hellozee) [reply .github/workflows/make_release.yml:35]: nope this is a wip

File: tetrateci/setup_boring_go.sh [hunk 5]
```diff
@@ -1,13 +1,28 @@
...
+  export GOLANG_VERSION=1.14.12b4
```
Comment (deva26) [inline tetrateci/setup_boring_go.sh:5]: Thinking loud, I think it is better that we refrain from mentioning the patch version.

Comment (hellozee) [reply tetrateci/setup_boring_go.sh:5]: Have to look at this, right now I don't have much idea how to work around that

**Review by deva26 — APPROVED** [review]

**Pull request merged**